File: prompts/fixtures/go/debug_test.go
package main

import "fmt"

func add(a, b int) int {
	result := a + b       // line 6
	return result          // line 7
}

func multiply(a, b int) int {
	result := a * b       // line 11
	return result          // line 12
}

func compute(x, y int) int {
	sum := add(x, y)              // line 16
	product := multiply(x, y)     // line 17
	final := sum + product        // line 18
	return final                   // line 19
}

func loopSum(n int) int {
	total := 0                     // line 23
	for i := 1; i <= n; i++ {
		total = add(total, i)     // line 25
	}
	return total                   // line 27
}

func factorial(n int) int {
	if n <= 1 {                    // line 31
		return 1                   // line 32
	}
	return n * factorial(n-1)     // line 34
}

func main() {
	x := 10                                // line 38
	y := 20                                // line 39
	result1 := compute(x, y)              // line 40
	fmt.Printf("compute = %d\n", result1) // line 41
	result2 := loopSum(5)                  // line 42
	fmt.Printf("loop_sum = %d\n", result2) // line 43
	result3 := factorial(5)                // line 44
	fmt.Printf("fact = %d\n", result3)    // line 45
}

File: prompts/fixtures/cpp/debug_test.cpp
#include <cstdio>

int add(int a, int b) {
    int result = a + b;        // line 4
    return result;              // line 5
}

int multiply(int a, int b) {
    int result = a * b;        // line 9
    return result;              // line 10
}

int compute(int x, int y) {
    int sum = add(x, y);              // line 14
    int product = multiply(x, y);     // line 15
    int final_ = sum + product;       // line 16
    return final_;                     // line 17
}

int loop_sum(int n) {
    int total = 0;                     // line 21
    for (int i = 1; i <= n; i++) {
        total = add(total, i);        // line 23
    }
    return total;                      // line 25
}

int factorial(int n) {
    if (n <= 1) return 1;             // line 29
    return n * factorial(n - 1);      // line 30
}

int main() {
    int x = 10;                                    // line 34
    int y = 20;                                    // line 35
    int result1 = compute(x, y);                  // line 36
    std::printf("compute = %d\n", result1);       // line 37
    int result2 = loop_sum(5);                     // line 38
    std::printf("loop_sum = %d\n", result2);      // line 39
    int result3 = factorial(5);                    // line 40
    std::printf("fact = %d\n", result3);          // line 41
    return 0;                                      // line 42
}
